qualang_tools/control_panel/video_mode/inner_loop_actions.py

```diff
@@ -0,0 +1,110 @@
+from abc import ABC, abstractmethod
+from typing import Tuple
+from qm.qua import declare, fixed, demod, set_dc_offset, align, wait, measure, QuaVariableType
+
+
+class InnerLoopAction(ABC):
+    @abstractmethod
+    def __call__(self, x: QuaVariableType, y: QuaVariableType) -> Tuple[QuaVariableType, QuaVariableType]:
+        pass
+
+    @abstractmethod
+    def initial_action(self):
+        pass
+
+
+class BasicInnerLoopAction(InnerLoopAction):
+    """Inner loop action for the video mode: set voltages and measure.
+
+    This class is responsible for performing the inner loop action for the video mode.
+    It is used to set the voltages and measure the readout pulse.
+
+    Args:
+        x_element: The name of the element along the x-axis to set the voltage.
+        y_element: The name of the element along the y-axis to set the voltage.
+        readout_element: The name of the element to measure.
+        readout_pulse: The name of the pulse to measure.
+        pre_measurement_delay: The delay before the measurement.
```

Comment:
Specify that the unit is ns (as opposed to clock cycles)

Reply:
fixed